src/coreclr/jit/lsrabuild.cpp

```diff
@@ -272,16 +269,11 @@ void LinearScan::resolveConflictingDefAndUse(Interval* interval, RefPosition* de
     }
     if (defRefPosition->isFixedRegRef && !defRegConflict)
     {
-        defReg       = defRefPosition->assignedReg();
-        defRegRecord = getRegisterRecord(defReg);
+        defReg = defRefPosition->assignedReg();
         if (canChangeUseAssignment)
         {
-            RefPosition* currFixedRegRefPosition = defRegRecord->recentRefPosition;
-            assert(currFixedRegRefPosition != nullptr &&
```

Comment:
any reason why this assert is removed?

Reply:
Let me readd it, I guess I removed it because we no longer need `defRegRecord` or `currFixedRegRefPosition` for anything, but I'll just add it back under DEBUG